lib/core/src/irods_client_negotiation.cpp

```diff
@@ -32,16 +32,15 @@ namespace irods
 {
     /// =-=-=-=-=-=-=-
     /// @brief given a property map and the target host name decide between a federated key and a local key
-    const std::string& determine_negotiation_key( const std::string& _host_name )
+    const std::string determine_negotiation_key( const std::string& _host_name )
     {
         // search the federation map for the host name
         try {
-            for ( const auto& el : irods::get_server_property<const std::vector<boost::any>>(irods::CFG_FEDERATION_KW) ) {
+            for ( const auto federation : irods::get_server_property<nlohmann::json>(irods::CFG_FEDERATION_KW) ) {
```

Comment:
Should `federation` be a reference?

Is the call to `get_server_property` producing a copy or a reference?